pkg/frontend/querymiddleware/codec.go

```diff
@@ -302,10 +345,14 @@ func (prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response, _ R
 	}
 	log.LogFields(otlog.Int("bytes", len(buf)))
 
+	start := time.Now()
 	if err := json.Unmarshal(buf, &resp); err != nil {
 		return nil, apierror.Newf(apierror.TypeInternal, "error decoding response: %v", err)
 	}
 
+	c.metrics.duration.WithLabelValues(operationDecode, formatJSON).Observe(time.Since(start).Seconds())
+	c.metrics.size.WithLabelValues(operationDecode, formatJSON).Observe(float64(len(buf)))
+
 	if resp.Status == statusError {
```

Comment:
Should the status of the result be a label in the histograms?

Reply:
I was trying to keep the cardinality as low as possible, but happy to add it in. 

Were you thinking of including the HTTP status code, or the `status` value embedded in the response payload (`"success"` or `"error"` etc.)?

Reply:
I'm also a bit concerned about cardinality. What value would give us the status code? What do we want to get from these metrics?

Reply:
I was thinking of the value of `status` - just enough so that we would be able to differentiate between the time taken to encode success response vs error responses. If there's no value in differentiating between them, fine to just omit the status.

Reply:
We see relatively few error responses, so I'm not sure there's much value in differentiating between them. We can always add it later if we need it.